### ISSUE TITLE
Link error when using ROSPACK_USE_DEPENDENCY

### ISSUE THREAD

**Issue by francois-keith** [jrl-umi3218/jrl-cmakemodules#20]
Hello,

There seems to be a link error related to the order of the link flags on Ubuntu 12.04 - 64bit when using the macro ROSPACK_USE_DEPENDENCY.
As a result, packages using this method do not link correctly.

This error appears for the package https://github.com/sbarthelemy/metapod/tree/add_metapodfromurdf
See https://github.com/laas/metapod/issues/44 for the issue declaration.
Since I think the following patch concerns all package using ros.cmale, I prefer starting the discussion here.

The link error appears whereas the link flags are present in the compilation line (but to not seem to be at the right place).
However, adding a TARGET_LINK_LIBRARY seems to do the trick though.

<pre>
FOREACH(dep roscpp rosconsole urdf)
  ADD_ROSPACK_DEPENDENCY(${dep})
  ROSPACK_USE_DEPENDENCY(metapodfromurdf ${dep})
  TARGET_LINK_LIBRARIES(metapodfromurdf ${dep}) # resolve the issue
ENDFOREACH()
</pre>

This correction patches the linkage for the specific package metapod, but I wonder if it wouldn't be better to correct all packages depending on ros by directly modifying the file jrl-cmake/ros.cmake
Indeed, this seems to be an issue similar to the one that occurred with the pkg-config dependencies.
In other words, similarly to these lines in the file cmake/pkg-config.cmake

<pre>
  IF(UNIX AND NOT APPLE)
    TARGET_LINK_LIBRARIES(${TARGET} ${${PREFIX}_LDFLAGS})
    TARGET_LINK_LIBRARIES(${TARGET} ${${PREFIX}_LDFLAGS_OTHER})
  ENDIF(UNIX AND NOT APPLE)
</pre>

I'd like to add in the cmake/ros.cmake file has the equivalent form, aka

<pre>
  IF(UNIX AND NOT APPLE)
    TARGET_LINK_LIBRARIES(${TARGET} ${PKG})
  ENDIF(UNIX AND NOT APPLE)
</pre>

Are you ok with that?

François
